httpbp/middlewares.go

```diff
@@ -510,3 +520,31 @@ func (rr *responseRecorder) WriteHeader(code int) {
 	rr.ResponseWriter.WriteHeader(code)
 	rr.responseCode = code
 }
+
+const (
+	RedditDebug        = "X-Reddit-Debug"
+	RedditDebugEnabled = "enabled"
+	RedditK8sMeta      = "X-Reddit-K8S-Meta"
+	redditK8sMetaFmt   = "%s-%s-%s"
+)
+
+// InjectDebugHeaders injects debug headers into the http response.
+// headers are only injected if X-Reddit-Debug is set in the corresponding request.
+func InjectDebugHeaders(metadata *metadatabp.Config) Middleware {
+	return func(name string, next HandlerFunc) HandlerFunc {
+		return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
+			wrapped := &statusCodeRecorder{ResponseWriter: w}
+			if value := r.Header.Get(RedditDebug); value == RedditDebugEnabled && metadata != nil {
```

Comment:
If what you want to do is be able to track down the logs for a request, I think a safer way to do that is to instead add a log tag into your context rather than returning k8s metadata to the client. Then you can search the logs for just that tag.